x/cronos/rpc/api.go

```diff
@@ -0,0 +1,238 @@
+package rpc
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/tendermint/tendermint/libs/log"
+	rpcclient "github.com/tendermint/tendermint/rpc/jsonrpc/client"
+	evmrpc "github.com/tharsis/ethermint/rpc"
+	"github.com/tharsis/ethermint/rpc/ethereum/backend"
+	rpctypes "github.com/tharsis/ethermint/rpc/ethereum/types"
+	ethermint "github.com/tharsis/ethermint/types"
+	evmtypes "github.com/tharsis/ethermint/x/evm/types"
+)
+
+const (
+	// CronosNamespace is the extension RPC namespace of cronos module.
+	CronosNamespace = "cronos"
+
+	apiVersion = "1.0"
+)
+
+func init() {
+	if err := evmrpc.RegisterAPINamespace(CronosNamespace, CreateCronosRPCAPIs); err != nil {
+		panic(err)
+	}
+}
+
+// CreateCronosRPCAPIs creates extension json-rpc apis
+func CreateCronosRPCAPIs(ctx *server.Context, clientCtx client.Context, tmWSClient *rpcclient.WSClient) []rpc.API {
+	evmBackend := backend.NewEVMBackend(ctx, ctx.Logger, clientCtx)
+	return []rpc.API{
+		{
+			Namespace: CronosNamespace,
+			Version:   apiVersion,
+			Service:   NewCronosAPI(ctx.Logger, clientCtx, evmBackend),
+			Public:    true,
+		},
+	}
+}
+
+// CronosAPI is the extension jsonrpc apis prefixed with cronos_.
+type CronosAPI struct {
+	ctx          context.Context
+	clientCtx    client.Context
+	queryClient  *rpctypes.QueryClient
+	chainIDEpoch *big.Int
+	logger       log.Logger
+	backend      backend.Backend
+}
+
+// NewCronosAPI creates an instance of the cronos web3 extension apis.
+func NewCronosAPI(
+	logger log.Logger,
+	clientCtx client.Context,
+	backend backend.Backend,
+) *CronosAPI {
+	eip155ChainID, err := ethermint.ParseChainID(clientCtx.ChainID)
+	if err != nil {
+		panic(err)
+	}
+	return &CronosAPI{
+		ctx:          context.Background(),
+		clientCtx:    clientCtx,
+		queryClient:  rpctypes.NewQueryClient(clientCtx),
+		chainIDEpoch: eip155ChainID,
+		logger:       logger.With("client", "json-rpc"),
+		backend:      backend,
+	}
+}
+
+// GetTransactionReceiptsByBlock returns all the transaction receipts included in the block.
+func (api *CronosAPI) GetTransactionReceiptsByBlock(blockNrOrHash rpctypes.BlockNumberOrHash) ([]map[string]interface{}, error) {
+	var receipts []map[string]interface{}
+
+	api.logger.Debug("cronos_getTransactionReceiptsByBlock", "blockNrOrHash", blockNrOrHash)
+
+	blockNum, err := api.getBlockNumber(blockNrOrHash)
+	if err != nil {
+		return nil, err
+	}
+
+	resBlock, err := api.clientCtx.Client.Block(api.ctx, blockNum.TmHeight())
+	if err != nil {
+		api.logger.Debug("block not found", "height", blockNum, "error", err.Error())
+		return nil, nil
+	}
+
+	blockNumber := resBlock.Block.Height
```

Comment:
We already had the blockNumber on line 86

Reply:
there's a difference though, the `blockNum` has the special values for latest/pending, but the `resBlock.Block.Height` is a exact number, I've changed the following logic to use this one.